### DISCUSSION TITLE
Would this theoretically work with AWS EC2?

### DISCUSSION THREAD

**Discussion by yagehu** [dep-sys/nix-dabei#9] [Q&A]
AWS requires you to specify an AMI image to boot with. Do you know if it's theoretically possible to re-provision an EC2 VM? If it is I would love to help with adding AWS EC2 support.

At a high level, what about the current code that is Hetzner-specific? I'm using Linode as a provider as well and wondering what it takes to support it.

**Comment by phaer** [comment]
Hi, there should be absolutely nothing hetzner specific inside this repo at the moment.

I plan to implement the first full provider support for hetzner by porting over previous work in https://github.com/dep-sys/nixos-zfs-installer, but I haven't done that yet.

I haven't tested Linode, but I believe that one should work pretty much out of the box. Just `nix build .#dist`, boot into the rescue system (or any other "live system"), copy ./results/* to that system and run kexec-boot. That should probably work if the rescue systems kernel supports kexec and if there are no necessary drivers missing. I'd be willing to assist here with Q&A if you run into trouble.

Same of AWS EC2, but i know even less about that one. I believe there's official NixOS images for EC2? I'd suggest looking into them to find out what it could take to support nix-dabei there.
